src/init.cpp

```diff
@@ -1373,9 +1373,16 @@ bool AppInitMain(InitInterfaces& interfaces)
     fDiscover = gArgs.GetBoolArg("-discover", true);
     fRelayTxes = !gArgs.GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY);
 
+    int listen_port = chainparams.GetDefaultPort();
```

Comment:
Should use `int64_t`, otherwise it might wrap with `GetArg`?

```suggestion
    int64_t listen_port = chainparams.GetDefaultPort();
```